File: dotnet-procapi/Preprocessors/InputPreprocessor.cs
using System;
using System.IO;
using ProcAPI.Contracts;
using SixLabors.ImageSharp;
using SixLabors.ImageSharp.Processing;

namespace ProcAPI.Preprocessors
{
    public class InputPreprocessor : IPreprocessor<InputData, string>
    {
        private const int Width = 256;
        private const int Height = 256;

        public string PreprocessData(InputData data)
        {
            // parse from base64
            var imgBytes = Convert.FromBase64String(data.img64);
            using var ms = new MemoryStream(imgBytes);
            using var img = Image.Load(ms);

            // resize
            img.Mutate(x =>
            {
                x.Resize(Width, Height);
                x.Grayscale();
            });

            // to base64
            using var oms = new MemoryStream();
            img.SaveAsJpeg(oms);
            var output64 = Convert.ToBase64String(oms.GetBuffer());

            return output64;
        }
    }
}

File: python-procapi/Preprocessors/InputPreprocessor.py
from Contracts.InputData import InputData
from Preprocessors.IPreprocessor import IPreprocessor
from base64 import b64decode, b64encode
from PIL import Image
import io


class InputPreprocessor(IPreprocessor):
    __width = 256
    __height = 256

    def preprocess_data(self, data: InputData) -> str:
        buf = io.BytesIO(
            b64decode(data.img64.encode())
        )
        img = Image.open(buf)
        img = img.resize((self.__width, self.__height)).convert(mode='L')

        buf = io.BytesIO()
        img.save(buf, format="JPEG")
        return b64encode(buf.getvalue())


